File: app/src/main/java/com/example/pijus/chemijosprojektas/losjonas2.java
package com.example.pijus.chemijosprojektas;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

public class losjonas2 extends AppCompatActivity {
    Button voniaskaiciuoti;
    EditText pirmasvonia;
    EditText antrasvonia;
    EditText treciasvonia;
    EditText ketvirtasvonia;
    EditText penktasvonia;
    TextView atsakymas;
    double aliejus=22;
    double vaskas=12;
    double lasas=1;
    double alavijo=5;
    double E=1;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_losjonas2);
        voniaskaiciuoti= findViewById(R.id.buttonl3);
        pirmasvonia= findViewById(R.id.ivestisl1);
        antrasvonia= findViewById(R.id.ivestisl2);
        treciasvonia= findViewById(R.id.ivestisl3);
        ketvirtasvonia= findViewById(R.id.ivestisl4);
        penktasvonia= findViewById(R.id.ivestisl5);
        atsakymas= findViewById(R.id.textViewl2);
        voniaskaiciuoti.setOnClickListener(v -> {
            atsakymas.setText("");
            if(pirmasvonia.getText().toString().isEmpty()||antrasvonia.getText().toString().isEmpty()||treciasvonia.getText().toString().isEmpty()||ketvirtasvonia.getText().toString().isEmpty()||penktasvonia.getText().toString().isEmpty()){
                Toast.makeText(getApplicationContext(),"Prašome užpildyti visus langelius", Toast.LENGTH_LONG).show();
            }
            else {
                double vanduo1 = Double.parseDouble(pirmasvonia.getText().toString());
                double vanduo2 = Double.parseDouble(antrasvonia.getText().toString());
                double vanduo4 = Double.parseDouble(ketvirtasvonia.getText().toString());
                double vanduo3 = Double.parseDouble(treciasvonia.getText().toString());
                double vanduo5 = Double.parseDouble(penktasvonia.getText().toString());
                double min = 100000;

                if (min > vanduo1/aliejus) {
                    min = vanduo1/aliejus;
                }
                if (min > vanduo2/vaskas) {
                    min = vanduo2/vaskas;
                }
                if (min > vanduo3/lasas) {
                    min = vanduo3/lasas;
                }
                if (min > vanduo4/alavijo) {
                    min = vanduo4/alavijo;
                }
                if (min > vanduo5/E) {
                    min = vanduo5/E;
                }
                atsakymas.setText(String.format("%.2f", min) + "porcijų");
            }
        });
    }
}

File: app/src/main/java/com/example/pijus/chemijosprojektas/PossibleAmount/losjonas2.kt
package com.example.pijus.chemijosprojektas.PossibleAmount

import android.os.Bundle
import com.example.pijus.chemijosprojektas.R

class losjonas2 : BaseCalculatePortionsActivity() {
    var aliejus = 22.0
    var vaskas = 12.0
    var lasas = 1.0
    var alavijo = 5.0
    var E = 1.0
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_losjonas2)
        calculateButton = findViewById(R.id.buttonl3)
        editTexts.add(findViewById(R.id.ivestisl1))
        editTexts.add(findViewById(R.id.ivestisl2))
        editTexts.add(findViewById(R.id.ivestisl3))
        editTexts.add(findViewById(R.id.ivestisl4))
        editTexts.add(findViewById(R.id.ivestisl5))
        ingredients.add(aliejus)
        ingredients.add(vaskas)
        ingredients.add(lasas)
        ingredients.add(alavijo)
        ingredients.add(E)

        answer = findViewById(R.id.textViewl2)
        calculateButton!!.setOnClickListener { calculatePortion() }
    }
}